sources/driverdog/src/main.rs

```diff
@@ -196,24 +214,26 @@ where
         dependencies_paths.push(object_file_path.to_string_lossy().into_owned());
     }
 
-    // Link the kernel module
-    let mut args = vec![
-        "-r".to_string(),
-        "-T".to_string(),
-        common_module_script_path.display().to_string(),
-        "--build-id".to_string(),
-        "-o".to_string(),
-        kernel_module_path
-            .to_str()
-            .context(error::InvalidModulePathSnafu {
-                path: &kernel_module_path,
-            })?
-            .to_string(),
-    ];
-    args.append(&mut dependencies_paths);
-
-    command(LD_BIN_PATH, &args)?;
-    info!("Linked {}", name);
+    if !dependencies_paths.is_empty() {
+        // Link the kernel module
+        let mut args = vec![
```

Comment:
Having a different type would help here, then you'd know that `link_kernel_module` could only be called for a `Linkable` and could keep this logic unchanged.